service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java

```diff
@@ -2,92 +2,16 @@
 import static org.apache.commons.lang3.ObjectUtils.isEmpty;
-import static uk.gov.hmcts.reform.fpl.enums.PartyType.INDIVIDUAL;
 
 @Service
 public class ChildrenService {
 
-    public List<Element<Child>> prepareChildren(CaseData caseData) {
-        List<Element<Child>> childCollection = new ArrayList<>();
-
-        if (caseData.getAllChildren().isEmpty()) {
-            childCollection.add(emptyElementWithPartyId());
-
-        } else if (caseData.getConfidentialChildren().isEmpty()) {
-            return caseData.getAllChildren();
-
-        } else {
-            caseData.getAllChildren().forEach(element -> {
-                if (element.getValue().containsConfidentialDetails()) {
-                    childCollection.add(getElementToAdd(caseData.getConfidentialChildren(), element));
-                } else {
-                    childCollection.add(element);
-                }
-            });
-        }
-        return childCollection;
-    }
-
-    // expands collection in UI. A value (in this case partyId) needs to be set to expand the collection.
-    private Element<Child> emptyElementWithPartyId() {
-        return ElementUtils.element(Child.builder()
-            .party(ChildParty.builder().partyId(randomUUID().toString()).build())
-            .build());
-    }
-
-    private Element<Child> getElementToAdd(List<Element<Child>> confidentialChildren, Element<Child> element) {
-        return confidentialChildren.stream()
-            .filter(confidentialChild -> confidentialChild.getId().equals(element.getId()))
-            .findFirst()
-            .orElse(element);
-    }
-
-    public List<Element<Child>> modifyHiddenValues(List<Element<Child>> children) {
-        return children.stream()
-            .map(element -> {
-                Child.ChildBuilder builder = Child.builder();
-
-                if (element.getValue().getParty().getPartyId() == null) {
-                    addHiddenValues(element, builder);
-                } else {
-                    builder.party(element.getValue().getParty().toBuilder().build());
-                }
-
-                if (element.getValue().containsConfidentialDetails()) {
-                    builder.party(element.getValue().getParty().toBuilder()
-                        .address(null)
-                        .telephoneNumber(null)
-                        .email(null)
-                        .build());
-                }
-
-                return Element.<Child>builder()
-                    .id(element.getId())
-                    .value(builder.build())
-                    .build();
-            })
-            .collect(toList());
-    }
-
-    private void addHiddenValues(Element<Child> element, Child.ChildBuilder builder) {
-        builder.party(element.getValue().getParty().toBuilder()
-            .partyId(randomUUID().toString())
-            .partyType(INDIVIDUAL)
-            .build());
-    }
-
     public String getChildrenLabel(List<Element<Child>> children) {
```

Comment:
Now all we need is a generic implementation of this in a helper class and we can basically get rid of this and the respondent service 😄

Reply:
yassss

Reply:
+1 for that, lets put it in a tech debt ticket